src/orderbook.ts

```diff
@@ -42,6 +50,40 @@ export class Orderbook {
     this.bids = new Map();
   }
 
+  static serializeOffers(offers: Map<number, Offer>): object {
+    const o: any = {};
+    offers.forEach((value, key) => { o[key] = value.serialize(); });
+    return o;
+  }
+
+  static deserializeOffers(o: any): Map<number, Offer> {
+    const offers = new Map();
+    for (let [key, value] of Object.entries(o)) {
+      offers.set(key, Offer.deserialize(value));
+    }
+    return offers;
+  }
+
+  serialize(): object {
+    return {
+      baseToken: this.baseToken,
+      quoteToken: this.quoteToken,
+      remainingFractionAfterFee: this.remainingFractionAfterFee.serialize(),
+      asks: Orderbook.serializeOffers(this.asks),
+      bids: Orderbook.serializeOffers(this.bids)
+    };
+  }
+
+  static deserialize(o: any): Orderbook {
+    const result = new Orderbook("", "");
```

Comment:
Can you pass base/quote token and remainingFractionAfterFee directly in the constructor so the properties can stay readonly?

Reply:
I can change the constructor to allow passing everything directly. But then we lose the old constructor which changes the fee before storing it. What do you prefer?

Reply:
Can't you just do something like:
```suggestion
    const result = new Orderbook(o.baseToken, o.baseToken, Fraction.deserialize(o.fee));
```

And then in serialize have:
```
      fee: this.remainingFractionAfterFee.sub(new Fraction(1, 1)).neg(),
```

Reply:
It would work but I dislike it because I want to serialize  and deserialize what is really there and not work around a transformation in the constructor.
Sadly typescript doen't seem to support multiple constructors.

Reply:
What about this:
```ts
type OrderbookOptions =
    { fee: Fraction } |
    { fee: undefined, remainingFractionAfterFee: Fraction }

class Orderbook {
  readonly baseToken: string;
  readonly quoteToken: string;
  readonly remainingFractionAfterFee: Fraction;

  private asks: Map<number, Offer>; // Mapping from price to cumulative offers at this point.
  private bids: Map<number, Offer>; // Mapping from price to cumulative offers at this point.
  
  constructor(
    baseToken: string,
    quoteToken: string,
    options: OrderbookOptions = { fee: new Fraction(1, 1000) },
  ) {
    this.baseToken = baseToken;
    this.quoteToken = quoteToken;
    this.remainingFractionAfterFee = options.fee ?
      new Fraction(1, 1).sub(options.fee) :
      options.remainingFractionAfterFee;
    this.asks = new Map();
    this.bids = new Map();
  }
}
```